### PR TITLE
Delete knowledge

### PR DESCRIPTION
**Pull request by alistairknight**
# **What this PR does / why we need it**:
*A brief description of changes being made*
*Link to the relevant work items: e.g: Relates to ADO Work Item AB#213700 and builds on #3376 (link to ADO Build ID URL)*

# **Special notes for your reviewer**
*Any specific actions or notes on review?*

# Testing
*Any relevant testing information and pipeline runs.*

# Checklist (please delete before completing or setting auto-complete)
- [ ] Story Work items associated (not Tasks)
- [ ] Successful testing run(s) link provided
- [ ] Title pattern should be `{work item number}: {title}`
- [ ] Description covers all the changes in the PR
- [ ] This PR contains documentation
- [ ] This PR contains tests
- [ ] Version number in **package.json** has been updated

# **How does this PR make you feel**:
gif

### CODE REVIEW

File: app/storage/knowledge-document-repo.js [hunk 1]
```diff
@@ -25,7 +25,15 @@ const listKnowledge = async (category = '', orderBy = 'lastModified ', orderByDi
     const metadata = mapMetadataToBase(blob.metadata)
 
     blob.metadata = metadata
-    if (category === '' || metadata.category === category) {
+    if (search !== '' && metadata.fileName) {
+      if (metadata.fileName.toLowerCase().indexOf(search.toLowerCase()) > -1) {
```
Comment (sfitz42) [inline app/storage/knowledge-document-repo.js]: Could you split this up a bit? Maybe create a new var to store the lower case file name?

File: app/routes/knowledge.js [hunk 2]
```diff
@@ -62,7 +64,7 @@ module.exports = [{
   },
   handler: async (request, h) => {
     try {
-      await updateKnowledgeMetadata(request.params.id, request.payload)
+      await updateKnowledgeMetadata(request.params.id, { id: request.params.id, ...request.payload })
```
Comment (sfitz42) [inline app/routes/knowledge.js]: Could you construct the object before calling the function?

**Review by sfitz42 — CHANGES_REQUESTED** [review]
Just a couple of changes. Also could you have a look into why the pipelines are failing?

File: app/storage/knowledge-document-repo.js (hunk 1 above)
Comment (alistairknight) [reply app/storage/knowledge-document-repo.js]: I've refactored this

File: app/routes/knowledge.js (hunk 2 above)
Comment (alistairknight) [reply app/routes/knowledge.js]: I've refactored this

**Review by sfitz42 — APPROVED** [review]
LGTM

**Pull request merged**